Handling add/remove to/from migration edge cases

guilherme opened the pull request:
Making sure the table name is parsed correctly when an add/remove column migration have 'from'  in the table name.

pixeltrix commented:
@guilherme thanks for the PR - what happens when you do `rails g migration RemoveOriginallyFromFromUsers` or `AddSentToToUsers` ?

guilherme commented:
@pixeltrix it works fine. I've just updated the PR with the scenario that I am having the issue.

I tried to generalize and I thought the  test was failing for that generic case but it wasn't, just double checked. My bad.

rafaelfranca reviewed (changes requested):
Tests are broken

lugray commented on activerecord/lib/rails/generators/active_record/migration/migration_generator.rb:
```diff
@@ -28,7 +28,7 @@ def create_migration_file
           @migration_template = "migration.rb"
           case file_name
-          when /^(add|remove)_.*_(?:to|from)_(.*)/
+          when /^(add)_.*_(?:to)_(.*)/, /^(remove)_.*?_(?:to|from)_(.*)/
```
The same change (and test) should be made to allow `migration = "remove_email_address_from_sent_to_user"`

The uncaptured groups can then be removed entirely:
```ruby
when /^(add)_.*_to_(.*)/, /^(remove)_.*_from_(.*)/
```

guilherme replied:
thanks for the feedback! I was trying to come up with a table name with "to" on the title but couldn't find it :)

guilherme committed:
Handling add/remove to/from migration edge cases

Making sure the table name is parsed correctly when an add/remove column migration have 'from'/'to' in the table name.

lugray commented on railties/test/generators/migration_generator_test.rb:
```diff
@@ -73,6 +84,17 @@ def test_remove_migration_with_attributes
     end
   end
 
+  def test_remove_migration_with_table_having_from_in_title
```
test_remove_migration_with_table_having_~from~**to**_in_title

guilherme replied:
👍
